pkg/api/types.go

```diff
@@ -1503,6 +1506,37 @@ type LocalVolumeSource struct {
 	Path string
 }
 
+// Represents storage that is managed by an external CSI volume driver
+type CSIPersistentVolumeSource struct {
+	// Driver is the name of the driver to use for this volume.
+	// Required.
+	Driver string
+
+	// VolumeHandle is the unique volume name returned by the CSI volume
+	// plugin’s CreateVolume to refer to the volume on all subsequent calls.
+	// Required.
+	VolumeHandle string
+
+	// Optional: MountSecretRef is a reference to the secret object containing
+	// sensitive information to pass to the CSI driver during NodePublish.
+	// This may be empty if no secret is required. If the secret object contains
+	// more than one secret, all secrets are passed.
+	// +optional
+	MountSecretRef *SecretReference
```

Comment:
did we remove secret refs in the proposal?

Reply:
Yes.  Master  branch does not have it : https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/container-storage-interface.md#proposed-api